helm/cluster/files/opt/bin/setup-apiserver-environment.sh

```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+env_file="/etc/apiserver-environment"
+
+cpus="$(grep -c ^processor /proc/cpuinfo)"
+memory="$(awk '/MemTotal/ { printf "%d \n", $2/1024/1024 }' /proc/meminfo)"
+
+declare -i cpus
+declare -i memory
+
+# convert to millicores
+cpus=$((cpus*1000))
+
+# subtract dedicated millicores (see 'systemReserved' and 'kubeReserved' fields in the kubelet config file)
+cpus=$((cpus-600))
```

Comment:
Could we make this more flexible? We're using different CPU's, would there be a way to figure out how many CPU's we have on the node and do a calculation based on the amount of CPUs?

Reply:
Not sure I understand your question, line 4 does exactly that: extracts the number of cpu (well, cores) available on the system.

Reply:
I mean the hardcoded 600

Reply:
Clarifying: we might want to make the systemReserved and kubeReserved configurable.
At the moment the values are hardcoded here (that 600) and here in the kubelet config:
https://github.com/giantswarm/cluster/blob/0ab4c58d0b4ad004ef6b335afe38ee83feddb368/helm/cluster/files/etc/kubelet-configuration.yaml#L14-L21

@nprokopic , if we wanted to have those two as helm values, what would be the correct template?
Something like `.Values.providerIntegration.controlPlane.kubeadmConfig.ignition.containerLinuxConfig.additionalConfig.systemReserved.cpu` and `.Values.providerIntegration.controlPlane.kubeadmConfig.ignition.containerLinuxConfig.additionalConfig.kubeReserved.cpu` ?
And eventually, where's the correct template where to define the values and give defaults if not defined?

Reply:
`.Values.providerIntegration` is supposed to be immutable provider-specific values, hard-coded in `values.schema.json` in cluster-<provider> apps.

I would put these for now here:

```
internal:
  advancedConfiguration:
    kubelet:
      kubeReserved: { ... }
      systemReserved: { ... }
```

Because those are customizable per cluster and have these disclaimers 😄:

```
For Giant Swarm internal use only, not stable, or not supported by UIs.

Advanced configuration of cluster components, to be configured by Giant Swarm staff only.
```

And it does seem that you should really know what you're doing if you are modifying kubelet's `kubeReserved` and/or `systemReserved` config.

Later if we want to make this available to customers we can move it to `.Values.global`.

Reply:
I'm slowly adding docs in every new PR, so it should be more clear soon where to put what.

Reply:
@nprokopic, added those values in the last commits and updated the schema and the generated docs and values (I fucked up and did a rebase instead of a ff merge from main, so github shows way more changes than it should, sorry :sweat:)

Reply:
Sorry for delay, looking into it now.